#! /usr/bin/env python

PACKAGE='visp_tracker'
import roslib; roslib.load_manifest(PACKAGE)

from math import pi

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Common
gen.add("angle_appear", double_t, 0, "Maximal angle value to consider an appearing face", 65.0, 0.0, 90.0)
gen.add("angle_disappear", double_t, 0, "Minimal angle value to consider that a face disappear", 75.0, 0.0, 100.0)

# Specific to vpMbEdgeTracker.

# Apparently unused attributes:
# - points to track
# - angle step
# - mask sign

gen.add("mask_size", int_t, 0, "mask size", 7, 0, 15)
gen.add("n_mask", int_t, 0, "number of masks", 180, 0, 500)

gen.add("range", int_t, 0, "range", 5, 0, 50)
gen.add("threshold", double_t, 0, "threshold", 2000., 0., 5000.)

gen.add("mu1", double_t, 0, "contrast continuity (mu1)", 0.5, 0., 1.)
gen.add("mu2", double_t, 0, "contrast continuity (mu2)", 0.5, 0., 1.)

gen.add("sample_step", double_t, 0, "sample step", 3., 1., 50.)
gen.add("ntotal_sample", int_t, 0, "total samples number", 1000, 0, 10000)

gen.add("strip", int_t, 0, "strip", 2, 0, 10)
gen.add("min_samplestep", double_t, 0, "min sample step", 4., 0., 50.)
gen.add("aberration", double_t, 0, "aberration", 2, 0, 10)
gen.add("init_aberration", double_t, 0, "init aberration", 5, 0, 10)

gen.add("lambda", double_t, 0, "lambda (mbt)", 1., 0., 1.)
gen.add("first_threshold", double_t, 0, "init threshold (mbt)", 0.4, 0., 1.)

exit(gen.generate(PACKAGE, "tracker_mbt_client", "ModelBasedSettingsEdge"))
